File: LeetCode/Math/Golang/Leetcode357-统计各位数字都不同的数字个数.go
package sbgo

func countNumbersWithUniqueDigits(n int) int {
	if n == 0 {
		return 1
	}
	if n == 1 {
		return 10
	}
	base, mul, res := 9, 9, 10
	for i := 0; i < n-1; i++ {
		base *= (mul - i)
		res += base
	}
	return res
}

File: LeetCode/Math/Python/Leetcode357-统计各位数字都不同的数字个数.py
class Solution:
    def countNumbersWithUniqueDigits(self, n: int) -> int:
        if n == 0: return 1
        if n == 1: return 10
        base, mul, res = 9, 9, 10
        for i in range(n - 1):
            base *= (mul - i)
            res += base
        return res
